consensus/ibft/validators/bls.go

```diff
@@ -0,0 +1,303 @@
+package validators
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/0xPolygon/polygon-edge/crypto"
+	"github.com/0xPolygon/polygon-edge/helper/hex"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/fastrlp"
+)
+
+type BLSPubKey []byte
+
+func (k BLSPubKey) String() string {
+	return hex.EncodeToHex(k[:])
+}
+
+func (k BLSPubKey) MarshalText() ([]byte, error) {
+	return []byte(k.String()), nil
+}
+
+func (k *BLSPubKey) Scan(src interface{}) error {
+	stringVal, ok := src.([]byte)
+	if !ok {
+		return errors.New("invalid type assert")
+	}
+
+	x, err := hex.DecodeHex(string(stringVal))
+	if err != nil {
+		return err
+	}
+
+	copy(*k, x)
+
+	return nil
+}
+
+type BLSValidator struct {
+	Address   types.Address
+	BLSPubKey BLSPubKey
+}
+
+func (v *BLSValidator) MarshalRLPWith(ar *fastrlp.Arena) *fastrlp.Value {
+	val := ar.NewArray()
+
+	val.Set(ar.NewBytes(v.Address[:]))
+
+	if len(v.BLSPubKey) == 0 {
+		val.Set(ar.NewNull())
+	} else {
+		val.Set(ar.NewBytes(v.BLSPubKey))
+	}
+
+	return val
+}
+
+func (v *BLSValidator) UnmarshalRLPFrom(p *fastrlp.Parser, vv *fastrlp.Value) error {
+	vals, err := vv.GetElems()
+	if err != nil {
+		return fmt.Errorf("mismatch of RLP type for CommittedSeal, expected list but found %s", vv.Type())
+	}
+
+	{
```

Comment:
Nitpick: maybe the wrapping scope braces can be removed ?

Reply:
Resolving because this code has changed already